### PR TITLE
Add class to wrap data splits

### PR DESCRIPTION
**Pull request by gabbard**

### CODE REVIEW

File: pegasus_wrapper/artifact.py [hunk 1]
```diff
@@ -106,16 +106,3 @@ def preexisting(
         value: _T, *, locator: Optional[Locator] = None
     ) -> "ValueArtifact[_T]":
         return ValueArtifact(value, depends_on=immutableset(), locator=locator)
-
-    @staticmethod
-    def computed(
```
Comment (gabbard) [inline pegasus_wrapper/artifact.py:111]: This was only used in the one test edited below.  It is not really necessary now given that the constructor can be used in the same way.

File: pegasus_wrapper/key_value.py [hunk 2]
```diff
@@ -164,6 +164,13 @@ def downsample(
     )
 
 
+@attrs(frozen=True, slots=True)
+class DataSplit:
+    train: KeyValueStore = attrib(validator=instance_of(KeyValueStore), kw_only=True)
+    dev: KeyValueStore = attrib(validator=instance_of(KeyValueStore), kw_only=True)
+    test: KeyValueStore = attrib(validator=instance_of(KeyValueStore), kw_only=True)
```
Comment (berquist) [inline pegasus_wrapper/key_value.py:171]: Will it be an issue if any of these ever need to not exist/be empty?

Comment (gabbard) [reply pegasus_wrapper/key_value.py:171]: I went back and forth on this. On the one hand it is nice to be able to have one or more of these be `Optional`; on the other hand, that pollutes user code with lots of checks for `None` when in context they are quite sure of what their splits are.  I decided on just having an empty `KeyValueStore`s if e.g. there isn't a dev set, but I could be convinced to go the other way.

Comment (berquist) [reply pegasus_wrapper/key_value.py:171]: I didn't know if the stores could be empty. In my opinion, if a `KeyValueStore` can be empty in the same way a normal container can, then this is a better solution, it just isn't clear to me from the rest of the file how that works.

Comment (gabbard) [reply pegasus_wrapper/key_value.py:202]: Does 0a379de7a42da0dde9ac4b469400a3acd79baa73 help?

Comment (berquist) [reply pegasus_wrapper/key_value.py:202]: Yes, I think so.

**Pull request merged**